lib/test_queue/runner.rb

```diff
@@ -334,9 +371,15 @@ def distribute_queue
       return if relay?
       remote_workers = 0
 
-      until @queue.empty? && remote_workers == 0
+      until @discovering_suites_pid.nil? && @queue.empty? && remote_workers == 0
         queue_status(@start_time, @queue.size, @workers.size, remote_workers)
 
+        # Make sure our discovery process is still doing OK.
+        if @discovering_suites_pid && Process.waitpid(@discovering_suites_pid, Process::WNOHANG) != nil
+          @discovering_suites_pid = nil
```

Comment:
> You can see how it looks in d3d3c57.

I see. It doesn't get cleaner because we still have to rummage around with the pid when we make sure the process is still alive. Thanks for trying it out, :+1: here.